src/Command/ValidateCommand.php

```diff
@@ -82,8 +91,62 @@ protected function doExecute(DependencyGraph $graph): int
             return 0;
         }
 
+        if ($this->input->getOption('generate-baseline')) {
+            $this->writeBaseline($graph, $violations);
+            $this->io->success(sprintf('Generated baseline containing %d ignored violations.', $violations->count()));
+        }
+
         $this->io->writeln('Try running the validator with the --autofix option to automatically fix most of the violations.');
 
         return 1;
     }
+
+    private function filterViolations(DependencyGraph $graph, ViolationCollection $violations): ViolationCollection
+    {
+        if (!file_exists($this->getBaselinePath($graph))) {
+            return $violations;
+        }
+
+        $contents = file_get_contents($this->getBaselinePath($graph));
+        if (false === $contents) {
+            throw new UnexpectedValueException('Could not read baseline.');
+        }
+
+        $ignored = Yaml::parse(file_get_contents($this->getBaselinePath($graph)) ?? '')['ignore'] ?? [];
```

Comment:
Unnecessary to retrieve the file contents again, as you already did on line 110.